pytorch_lightning/plugins/training_type/single_tpu.py

```diff
@@ -86,9 +85,3 @@ def save_checkpoint(self, checkpoint: Dict[str, Any], filepath: _PATH) -> None:
     def teardown(self) -> None:
         # TPU teardown
         os.environ.pop("PT_XLA_DEBUG", None)
-
-    @SingleDevicePlugin.checkpoint_io.setter
-    def checkpoint_io(self, io: Optional[XLACheckpointIO]) -> None:
-        if io is not None and not isinstance(io, XLACheckpointIO):
-            raise MisconfigurationException(f"{self.__class__.__name__}.checkpoint_io` must be a `XLACheckpointIO`.")
-        self._checkpoint_io = io
```

Comment:
After we remove this, if user pass in a wrong/customized checkpoint, will we override it or fail silently?
Do you think we should include this logic in accelerator_connector? as plugins choosing and misconfig check should be in accelerator_connector?

Reply:
If the users chooses to do this, he should be allowed. It's his/her responsibility to make sure their implementation is correct.

Typing checks can help with what are the expected input types.

I don't think this validation should be moved.